lib/harvester.js

```diff
@@ -310,6 +309,40 @@ Harvester.prototype.after = function (name, fn) {
   return this;
 };
 
+Harvester.prototype.exportRoles = function () {
+    var roles = {};
+    _.forEach(this.createdResources, function (resource) {
+        _.forEach(resource.roles, function (role) {
+            roles[role] = roles[role] || [];
+            route.getRestMethods().forEach(function (httpMethod) {
+                if (resource[httpMethod]().options.notAllowed) {
+                        return;
+                    }
+                    var methodRoles = resource[httpMethod]().roles;
+                    var permission = resource[httpMethod]().getPermission();
+                    if (_.isEmpty(methodRoles)) {
+                        roles[role].push(permission);
+                    } else {
+                        _.forEach(methodRoles, function (role) {
+                            roles[role] = roles[role] || [];
+                            roles[role].push(permission);
+                        })
+                    }
+                });
+        })
+    });
+    _.forEach(roles, function (permissions, role) {
+        roles[role] = _.uniq(permissions);
+    });
+    return roles;
+};
+
+Harvester.prototype.roles = function () {
+    var roles = Array.prototype.slice.call(arguments);
+    this.createdResources[this._resource].roles = roles;
```

Comment:
this._resource should be constantly changing, so in cases where you call it, in general you want to also allow a string as param 0, which can be the resource name. On the otherhand, allowing this flexibility does make things more complicated. I think we're moving away from complicated things, so perhaps we just leave it like this.